crates/rover-client/src/query/partial/fetch.rs

```diff
@@ -0,0 +1,196 @@
+use crate::blocking::StudioClient;
+use crate::RoverClientError;
+use graphql_client::*;
+
+#[derive(GraphQLQuery)]
+// The paths are relative to the directory where your `Cargo.toml` is located.
+// Both json and the GraphQL schema language are supported as sources for the schema
+#[graphql(
+    query_path = "src/query/partial/fetch.graphql",
+    schema_path = ".schema/schema.graphql",
+    response_derives = "PartialEq, Debug, Serialize, Deserialize",
+    deprecated = "warn"
+)]
+/// This struct is used to generate the module containing `Variables` and
+/// `ResponseData` structs.
+/// Snake case of this name is the mod name. i.e. fetch_subgraph_query
+pub struct FetchSubgraphQuery;
+
+/// The main function to be used from this module. This function fetches a
+/// schema from apollo studio and returns it in either sdl (default) or json format
+pub fn run(
+    variables: fetch_subgraph_query::Variables,
+    client: &StudioClient,
+    // we can't specify this as a variable in the op, so we have to filter the
+    // operation response by this name
+    service_name: &str,
+) -> Result<String, RoverClientError> {
+    let graph_name = variables.graph_id.clone();
+    let response_data = client.post::<FetchSubgraphQuery>(variables)?;
+    let services = get_services_from_response_data(response_data, &graph_name)?;
```

Comment:
I am noticing that you pass a `&str` to the `get_services_from_response_data` that then gets converted to `to_string()`. How do you feel about changing the parameters of `get_services_from_response_data` to accept an owned value right away?

Reply:
Ah I see, that's not possible since our RoverClientError accepts only an owned string. Ignore the above comment then. What you have works!